Chef/AddRelativityVersionToSolutionSnapshotDatabase.ps1

```diff
@@ -0,0 +1,96 @@
+Clear-Host
+
+[string] $salesforceUsername = $args[0]
+[string] $salesforcePassword = $args[1]
+[string] $relativityVersion = $args[2]
+[string] $relativityVersionReleaseName = ''
+
+[string] $server = 'solutionsnapshotapi.azurewebsites.net'
+
+[string] $CreateRelativityVersionAsyncUrl = "https://$($server)/api/external/CreateRelativityVersionAsync"
+[string] $ReadRelativityVersionAsyncUrl = "https://$($server)/api/external/ReadRelativityVersionAsync"
+
+$global:salesforceSessionObject = $null
+
+function Write-Empty-Message() {
+  Write-Host ""
+}
+
+function Write-Method-Call-Message($message) {
+  Write-Empty-Message
+  Write-Host -ForegroundColor Cyan $message
+}
+
+function Write-Message($message) {
+  Write-Empty-Message
+  Write-Host -ForegroundColor Green $message
+}
+
+function Write-Error-Message($message) {
+  Write-Empty-Message
+  Write-Host -ForegroundColor Red $message
+}
+
+function GetSessionId() {
+  try {
+    [Reflection.Assembly]::LoadFile("$PSScriptRoot\SolutionSnapshotSalesforceLoginHelper.dll")
+    $salesforceSessionHelper = New-Object SolutionSnapshotSalesforceLoginHelper.SalesforceSessionHelper
+	  
+    Write-Method-Call-Message "Calling method to get SalesforceSessionInfo"
+    $salesforcesessioninfo = $salesforceSessionHelper.GetSalesforceSessionInfo($salesforceUsername, $salesforcePassword)
+    $global:salesforceSessionObject = new-object psobject
+    $global:salesforceSessionObject | add-member noteproperty salesforceuserid $salesforcesessioninfo.salesforceuserid
+    $global:salesforceSessionObject | add-member noteproperty sessionid $salesforcesessioninfo.sessionid
+    $global:salesforceSessionObject | add-member noteproperty serverurl $salesforcesessioninfo.serverurl
+  }
+  catch {
+    Write-Error-Message "An error occured when retrieving SalesforceSessionInfo."    
+    Write-Error-Message "Error Message: ($_)"
+  }
+}
+
+function CreateRelativityVersionAsync() {
+  try {
+    Write-Method-Call-Message "Calling CreateRelativityVersionAsync API"
+    $request = new-object psobject
+    $request | add-member noteproperty SalesforceSessionInfo $global:salesforceSessionObject
+    $request | add-member noteproperty Version $relativityVersion
+    $request | add-member noteproperty ReleaseName $relativityVersionReleaseName
+    
+    $requestJson = $request | ConvertTo-Json
+    Write-Message "Request Json: $($requestJson)"
+
+    $response = invoke-restmethod -uri $createrelativityversionasyncurl -method post -body $requestjson -contenttype 'application/json' -headers @{"x-csrf-header"="-"}
+    $responsejson = $response | convertto-json
+    Write-Message "Created Relativity Version"
+	Send-Slack-Success-Message $relativityVersion
+  }
+  catch {
+	if($_.ToString().Contains('Relativity Version ' + $relativityVersion + ' already exists')){
```

Comment:
we need a slack message when there is an exception

Reply:
Done

Reply:
Done

Reply:
Shouldn't the name of the function change to Send-Slack-Message (not necessary success)
Also I don't see the 2nd call to the slack function for error. Am I missing something?